graf3d/eve7/inc/ROOT/REveBox.hxx

```diff
@@ -0,0 +1,97 @@
+// @(#)root/eve:$Id$
+// Author: Matevz Tadel, 2010
+
+/*************************************************************************
+ * Copyright (C) 1995-2007, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_REveBox
+#define ROOT_REveBox
+
+#include "ROOT/REveShape.hxx"
+
+namespace ROOT {
+namespace Experimental {
+
+//------------------------------------------------------------------------------
+// REveBox
+//------------------------------------------------------------------------------
+
+class REveBox : public REveShape,
+                public REveProjectable
+{
+private:
+   REveBox(const REveBox&);            // Not implemented
+   REveBox& operator=(const REveBox&); // Not implemented
+
+protected:
+   Float_t fVertices[8][3];
+
+public:
+   REveBox(const char* n="REveBox", const char* t="");
+   virtual ~REveBox();
+
+   void SetVertex(Int_t i, Float_t x, Float_t y, Float_t z);
+   void SetVertex(Int_t i, const Float_t* v);
```

Comment:
Could we use `std::array<float, 3>` here? It will make the interface less error prone compared to C-style arrays

Reply:
@rete If it is OK, I would like to change this in a separate PR.